pandas/tests/scalar/test_period.py

```diff
@@ -1038,6 +1038,24 @@ def test_add_raises(self):
         with tm.assert_raises_regex(TypeError, msg):
             dt1 + dt2
 
+    boxes = [lambda x: x, lambda x: pd.Series([x]), lambda x: pd.Index([x])]
+
+    @pytest.mark.parametrize('lbox', boxes)
+    @pytest.mark.parametrize('rbox', boxes)
+    def test_add_timestamp_raises(self, rbox, lbox):
+        # GH # 17983
+        ts = pd.Timestamp('2017')
+        per = pd.Period('2017', freq='M')
+
+        with pytest.raises(TypeError):
```

Comment:
Let's make sure that the right error message is hit (use `tm.assert_raises_regex` to be safe, unclear ATM whether we're going to leverage the most recent `pytest` API).